packages/invoices/src/OutboundShipment/api/api.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  //   InsertOutboundShipmentLineInput,
+  //   DeleteOutboundShipmentLineInput,
+  //   UpdateOutboundShipmentLineInput,
+  RecordPatch,
+  OmSupplyApi,
+  UpdateOutboundShipmentInput,
+  InvoiceNodeStatus,
+  UpdateOutboundShipmentStatusInput,
+  NameResponse,
+  InvoicePriceResponse,
+  InvoiceQuery,
+  InvoiceLineConnector,
+  ConnectorError,
+  StockLineResponse,
+  StockLineNode,
+  LocationResponse,
+  ItemNode,
+  ItemResponse,
+} from '@openmsupply-client/common';
+import { Location } from '@openmsupply-client/system';
+import { Invoice, InvoiceLine } from '../../types';
+
+const otherPartyGuard = (otherParty: NameResponse) => {
+  if (otherParty.__typename === 'NameNode') {
+    return otherParty;
+  } else if (otherParty.__typename === 'NodeError') {
+    throw new Error(otherParty.error.description);
+  }
+
+  throw new Error('Unknown');
+};
+
+const pricingGuard = (pricing: InvoicePriceResponse) => {
+  if (pricing.__typename === 'InvoicePricingNode') {
+    return pricing;
+  } else if (pricing.__typename === 'NodeError') {
+    throw new Error(pricing.error.description);
+  } else {
+    throw new Error('Unknown');
+  }
+};
+
+const invoiceGuard = (invoiceQuery: InvoiceQuery) => {
+  if (invoiceQuery.invoice.__typename === 'InvoiceNode') {
+    return invoiceQuery.invoice;
+  }
+
+  throw new Error(invoiceQuery.invoice.error.description);
+};
+
+const linesGuard = (invoiceLines: InvoiceLineConnector | ConnectorError) => {
+  if (invoiceLines.__typename === 'InvoiceLineConnector') {
+    return invoiceLines.nodes;
+  }
+
+  if (invoiceLines.__typename === 'ConnectorError') {
+    throw new Error(invoiceLines.error.description);
+  }
+
+  throw new Error('Unknown');
+};
+
+const stockLineGuard = (stockLine: StockLineResponse): StockLineNode => {
+  if (stockLine.__typename === 'StockLineNode') {
+    return stockLine;
+  }
+
+  throw new Error('Unknown');
+};
+
+const locationGuard = (location: LocationResponse): Location => {
+  if (location.__typename === 'LocationNode') {
+    return location;
+  }
+
+  throw new Error('Unknown');
+};
+
+const itemGuard = (item: ItemResponse): ItemNode => {
+  if (item.__typename === 'ItemNode') {
+    return item;
+  }
+
+  throw new Error('Unknown');
```

Comment:
I wanted to make a `transform.ts` or something like that. However we run into a 'type' problem where determining the exact return value can often be a little murky. For instance I have here that this fn returns an `itemNode`, but that's because I have added in all the extra fields like `availableBatches` to the query.. if it wasn't a 'full' `ItemNode` then would have to create new types to add to the return value. 

Really just another thing to sort out soon with fragments

For instance we could create fragments like this,

```
fragment PartialItemNode {
    id
    code
    node
}.  
```

and use them in our queries, then we could generate types...

Though in saying that, hopefully one day we will have a less .. 'hardwork' api?